### PR TITLE
[resotocore][chore] Adjust analytics events

### PR DESCRIPTION
**Pull request by anjafr**
# Description
- event `system.installed` now with information about the installation method
- new event `system.configuration-deleted`
- new event `system.configuration-changed` with information about the configured providers and authentication methods

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [x] Document new or updated functionality (someengineering/resoto.com#412)

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

**Comment by TheCatLady** [comment]
[This documentation page](https://resoto.com/docs/edge/reference/telemetry) should probably be updated to reflect these changes. 😊

**Review by aquamatthias — COMMENTED** [review]
Sorry - I realized I did not press Submit Review...

**Comment by anjafr** [comment]
> [This documentation page](https://resoto.com/docs/edge/reference/telemetry) should probably be updated to reflect these changes. 😊

Good point, I'm on it!

File: resotocore/resotocore/core_config.py [hunk 1]
```diff
@@ -62,6 +62,28 @@ def inside_docker() -> bool:
...
+    )
+
+
+def helm_installation() -> bool:
+    """
+    Try to detect if we were installed via helm chart.
+    """
+    return (
+        # environment variables have to be set explicitly
+        os.environ.get("HELM_HELM_VERSION", "false")
```
Comment (aquamatthias) [inline resotocore/resotocore/core_config.py:82]:
```suggestion
        os.environ.get("HELM_VERSION", "false")
```

File: resotocore/resotocore/core_config.py [hunk 2]
```diff
@@ -62,6 +62,28 @@ def inside_docker() -> bool:
     )
 
 
+def inside_kubernetes() -> bool:
+    """
+    Try to detect if we are running on kubernetes.
+    """
+    return (
+        # environment variables have to be set explicitly
+        os.environ.get("HELM_HELM_VERSION", "false") != "false"
```
Comment (aquamatthias) [inline resotocore/resotocore/core_config.py:71]:
```suggestion
        os.environ.get("HELM_VERSION", "false") != "false"
```

File: resotocore/resotocore/core_config.py [hunk 3]
```diff
@@ -62,6 +62,28 @@ def inside_docker() -> bool:
...
+        or os.environ.get("KUBERNETES_SERVICE_HOST", "false") != "false"
```
Comment (aquamatthias) [inline resotocore/resotocore/core_config.py:72]: I would check for any env var that starts with `KUBERNETES_`

**Review by aquamatthias — COMMENTED** [review]
Thanks. Some small comments

File: resotocore/resotocore/config/__init__.py [hunk 4]
```diff
@@ -17,6 +19,68 @@ class ConfigEntity:
...
+            ):
+                analytics.update({"aws_use_access_secret_key": True})
+            if value_in_path(self.config, ["resotoworker", "aws", "role"]):
+                analytics.update({"aws_use_role": True})
+            if value_in_path(self.config, ["resotoworker", "aws", "profiles"]):
+                analytics.update({"aws_use_profiles": True})
+            if value_in_path(self.config, ["resotoworker", "aws", "account"]):
+                analytics.update({"aws_use_accounts": True})
+            if value_in_path(self.config, ["resotoworker", "aws", "scrape_org"]):
+                analytics.update({"aws_use_scrape_org": True})
```
[the remark this answers is not in the document]
Comment (anjafr) [reply resotocore/resotocore/config/__init__.py:53]: Wouldn't this put the actual keys in our data?!

[the remark this answers is not in the document]
Comment (aquamatthias) [reply resotocore/resotocore/config/__init__.py:53]: Sorry I don't want the keys, I wanted to propose to set the values in one step and then use it in the dict.
Tha'ts way it is easy to see where the value is coming from.

[the remark this answers is not in the document]
Comment (anjafr) [reply resotocore/resotocore/config/__init__.py]: Ok, got it. PTAL ⬇️ 😊

File: resotocore/resotocore/core_config.py [hunk 5]
```diff
@@ -62,6 +62,28 @@ def inside_docker() -> bool:
...
+
+
+def helm_installation() -> bool:
+    """
+    Try to detect if we were installed via helm chart.
+    """
+    return (
+        # environment variables have to be set explicitly
+        os.environ.get("HELM_VERSION", "false")
+        != "false"
```
Comment (aquamatthias) [inline resotocore/resotocore/core_config.py:83]:
```suggestion
        "HELM_VERSION" in os.environ
```
Not wrong, but maybe this one is easier to read?

File: resotocore/resotocore/config/__init__.py [hunk 6]
```diff
@@ -17,6 +19,66 @@ class ConfigEntity:
...
+            aws_use_access_secret_key = (
+                value_in_path(self.config, ["resotoworker", "aws", "access_key_id"]) is not None
+                and value_in_path(self.config, ["resotoworker", "aws", "secret_access_key"]) is not None
+            )
+            aws_use_role = value_in_path(self.config, ["resotoworker", "aws", "role"]) is not None
+            aws_use_profiles = value_in_path(self.config, ["resotoworker", "aws", "profiles"]) is not None
+            aws_use_accounts = value_in_path(self.config, ["resotoworker", "aws", "account"]) is not None
+            aws_use_scrape_org = value_in_path(self.config, ["resotoworker", "aws", "scrape_org"])
+
+            analytics.update({"aws_use_access_secret_key": aws_use_access_secret_key})
```
Comment (aquamatthias) [inline resotocore/resotocore/config/__init__.py:47]:
```suggestion
            analytics["aws_use_access_secret_key"] = aws_use_access_secret_key
```
This way no intermediate objects are created and is simpler to read?
True for all other `update` calls.

File: resotocore/resotocore/core_config.py [hunk 7]
```diff
@@ -62,6 +62,28 @@ def inside_docker() -> bool:
     )
 
 
+def inside_kubernetes() -> bool:
+    """
+    Try to detect if we are running on kubernetes.
+    """
+    return (
+        # environment variables have to be set explicitly
+        os.environ.get("HELM_VERSION", "false") != "false"
```
Comment (aquamatthias) [inline resotocore/resotocore/core_config.py:71]: Same here

**Review by aquamatthias — APPROVED** [review]
2 tiny code beautify requests - otherwise lgtm

**Review by aquamatthias — APPROVED** [review]
Thanks!

**Pull request merged**